src/main/java/hudson/plugins/promoted_builds/LastBuildPromotionStatusColumn.java

```diff
@@ -0,0 +1,59 @@
+package hudson.plugins.promoted_builds;
+
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import hudson.Extension;
+import hudson.model.Item;
+import hudson.model.Job;
+import hudson.model.ListView;
+import hudson.model.Run;
+import hudson.views.ListViewColumn;
+import hudson.views.ListViewColumnDescriptor;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Provides a last build promotion summary column for {@link ListView}s.
+ * @since TODO
+ */
+public class LastBuildPromotionStatusColumn extends ListViewColumn {
+
+    @DataBoundConstructor
+    public LastBuildPromotionStatusColumn() {
+        super();
+    }
+
+    public List<String> getPromotionIcons(final Item item) {
+        List<String> icons = new ArrayList<String>();
+        if (item instanceof Job<?, ?>) {
+            final Job<?, ?> job = (Job<?, ?>) item;
+            final Run<?, ?> b = job.getLastBuild();
+            PromotedBuildAction a = b != null ? b.getAction(PromotedBuildAction.class) : null;
+            if (a != null) {
+                for (Status s : a.getPromotions()) {
+                    icons.add(s.getIcon("16x16"));
+                }
+            }
+        }
+        return icons;
+    }
```

Comment:
NIT: delay creating of `icons` until needed, returning an `emptyList` if not needed.